src/server/scripts/poller.js

```diff
@@ -19,54 +20,81 @@ import requestGitHub from '../helpers/github';
 const logger = logging.getLogger('poller');
 raven.config(conf.get('SENTRY_DSN')).install();
 
-
 // Process all Repository (DB) models synchronously. Check for changes using
 // `checkSnapRepository` and if changed request a LP snap build and mark
 // it as 'updated'.
-export const pollRepositories = (checker, builder) => {
+export const pollRepositories = (checker) => {
   logger.info('GitHub Repository Poller ...');
   var raven_client = raven.Client(conf.get('SENTRY_DSN'));
 
   checker = checker || checkSnapRepository;
-  builder = builder || buildSnapRepository;
 
   // XXX: meh! no ES6 import support, great library :-/
   let AsyncLock = require('async-lock');
   let pollRepoLock = new AsyncLock();
+  let locked_promises = [];
 
   return db.model('Repository').fetchAll().then(function (results) {
     logger.info(`Iterating over ${results.length} repositories.`);
     results.models.forEach((repo) => {
-      pollRepoLock.acquire('PROCESS-REPO-SYNC', async () => {
+      const p = pollRepoLock.acquire('PROCESS-REPO-SYNC', async () => {
         const owner = repo.get('owner');
         const name = repo.get('name');
+        const store_name = repo.get('store_name');
```

Comment:
I know that not reading from db is not in scope of this PR, but just a reminder that couple lines later we do `const snap = await internalFindSnap(repositoryUrl);` that would give us a more up to date value of `store_name`.